jjb/include/pipelines/qpc-tests-pipeline.groovy

```diff
@@ -0,0 +1,529 @@
+        sh 'sudo docker ps -a'
+        sh 'sudo docker logs quipucords | grep -i "Running without supervisord"'
+    }}
+}}
+
+
+def containerInstall(distro) {{
+    echo "Execute docker container install"
+    def qpc_version = getQPCVersion()
+    if (distro == 'rhel6') {{
+        echo 'No systemd, Docker should already be started.'
+    }} else {{
+        echo 'Starting Docker using Systemd'
+        sh 'sudo systemctl start docker'
+        sh 'sudo systemctl status docker'
+    }}
+
+    sh """\
+    sudo docker load -i quipucords.${{qpc_version}}.tar.gz
+    # make log dir to save server logs
+    mkdir -p log
+    """.stripIndent()
+}}
+
+
+def installQpcClient(distro) {{
+    // Install the qpc client
+    echo 'Install QPC Client'
+    sh '''\
+    sudo wget -O /etc/yum.repos.d/group_quipucords-qpc-fedora-28.repo https://copr.fedorainfracloud.org/coprs/g/quipucords/qpc/repo/fedora-28/group_quipucords-qpc-fedora28.repo
+    ls -la /etc/yum.repos.d/
+    '''.stripIndent()
+
+    if (distro ==~ /f\d\d/) {{
+        sh 'sudo dnf -y install qpc'
+    }} else {{
+        sh 'sudo yum -y install qpc'
+    }}
+}}
+
+
+
+////////////////////
+//// Test Setup ////
+////////////////////
+def setupScanUsers() {{
+    dir('ci') {{
+        git 'https://github.com/quipucords/ci.git'
+    }}
+
+    sshagent(['390bdc1f-73c6-457e-81de-9e794478e0e']) {{
+        withCredentials([file(credentialsId: '50dc19ce-555f-422c-af38-3b5ede422bb4', variable: 'ID_JENKINS_RSA_PUB')]) {{
+            sh 'sudo dnf -y install ansible'
+
+            sh '''\
+            cat > jenkins-slave-hosts <<EOF
+            [jenkins-slave]
+            ${{OPENSTACK_PUBLIC_IP}}
+
+            [jenkins-slave:vars]
+            ansible_user=jenkins
+            ansible_ssh_extra_args=-o StrictHostKeyChecking=no
+            ssh_public_key_file=$(cat ${{ID_JENKINS_RSA_PUB}})
+            EOF
+            '''.stripIndent()
+
+            sh 'ansible-playbook -b -i jenkins-slave-hosts ci/ansible/sonar-setup-scan-users.yaml'
+        }}
+    }}
+}}
+
+
+def setupCamayoc() {{
+    dir('camayoc') {{
+        git 'https://github.com/quipucords/camayoc.git'
+    }}
+
+    sh '''\
+    sudo pip install ./camayoc[dev]
+    cp camayoc/pytest.ini .
+    '''.stripIndent()
+
+    withCredentials([file(credentialsId: '4c692211-c5e1-4354-8e1b-b9d0276c29d9', variable: 'ID_JENKINS_RSA')]) {{
+        sh '''\
+        mkdir -p /home/jenkins/.ssh
+        cp "${{ID_JENKINS_RSA}}" /home/jenkins/.ssh/id_rsa
+        chmod 0600 /home/jenkins/.ssh/id_rsa
+        '''.stripIndent()
+    }}
+
+    configFileProvider([configFile(fileId: '62cf0ccc-220e-4177-9eab-f39701bff8d7', targetLocation: 'camayoc/config.yaml')]) {{
+        sh '''\
+        sed -i "s/{{jenkins_slave_ip}}/${{OPENSTACK_PUBLIC_IP}}/" camayoc/config.yaml
+        '''.stripIndent()
+
+    }}
+}}
+
+
+
+////////////////////////
+//// Test Functions ////
+////////////////////////
+def runInstallTests(distro) {{
+    if (distro ==~ /f\d\d/) {{
```

Comment:
You are checking for the distro to install packages on some places, you may consider creating a function for that in the future.

Reply:
Good call. I'm looking to fix up the install tests soon, and will keep that in mind during the rework here...